scrapy/loader/__init__.py

```diff
@@ -3,219 +3,104 @@
-    return method
+    Return a new Item Loader for populating the given Item. If no item is
+    given, one is instantiated automatically using the class in
+    :attr:`default_item_class`.
+
+    When instantiated with a ``selector`` or a ``response`` parameters
+    the :class:`ItemLoader` class provides convenient mechanisms for extracting
+    data from web pages using :ref:`selectors <topics-selectors>`.
+
+    :param item: The item instance to populate using subsequent calls to
+        :meth:`~ItemLoader.add_xpath`, :meth:`~ItemLoader.add_css`,
+        or :meth:`~ItemLoader.add_value`.
+    :type item: :class:`~scrapy.item.Item` object
+
+    :param selector: The selector to extract data from, when using the
+        :meth:`add_xpath` (resp. :meth:`add_css`) or :meth:`replace_xpath`
+        (resp. :meth:`replace_css`) method.
+    :type selector: :class:`~scrapy.selector.Selector` object
+
+    :param response: The response used to construct the selector using the
+        :attr:`default_selector_class`, unless the selector argument is given,
+        in which case this argument is ignored.
+    :type response: :class:`~scrapy.http.Response` object
+
+    The item, selector, response and the remaining keyword arguments are
+    assigned to the Loader context (accessible through the :attr:`context` attribute).
+
+    .. attribute:: item
+
+        The item object being parsed by this Item Loader.
+        This is mostly used as a property so when attempting to override this
+        value, you may want to check out :attr:`default_item_class` first.
+
+    .. attribute:: context
+
+        The currently active :ref:`Context <loaders-context>` of this Item Loader.
+        Refer to <loaders-context> for more information about the Loader Context.
+
+    .. attribute:: default_item_class
 
+        An Item class (or factory), used to instantiate items when not given in
+        the ``__init__`` method.
 
-class ItemLoader:
+    .. attribute:: default_input_processor
+
+        The default input processor to use for those fields which don't specify
+        one.
+
+    .. attribute:: default_output_processor
+
+        The default output processor to use for those fields which don't specify
+        one.
+
+    .. attribute:: default_selector_class
+
+        The class used to construct the :attr:`selector` of this
+        :class:`ItemLoader`, if only a response is given in the ``__init__`` method.
+        If a selector is given in the ``__init__`` method this attribute is ignored.
+        This attribute is sometimes overridden in subclasses.
+
+    .. attribute:: selector
+
+        The :class:`~scrapy.selector.Selector` object to extract data from.
+        It's either the selector given in the ``__init__`` method or one created from
+        the response given in the ``__init__`` method using the
+        :attr:`default_selector_class`. This attribute is meant to be
+        read-only.
+    """
 
     default_item_class = Item
```

Comment:
Is this useful? I ask because I imagine that, given `Item` has no fields defined, this always needs to be overriden with a different value.

Reply:
This is here only for backwards compatibility, as this is how it was before.
I guess there should be no harm in removing it

Reply:
Right, backward compatibility. Keeping it makes sense.